devicetypes/smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy

```diff
@@ -98,6 +98,14 @@ def parse(String description) {
 				} else {
 					log.warn "TEMP REPORTING CONFIG FAILED- error code: ${descMap.data[0]}"
 				}
+			} else if (descMap?.clusterInt == 0x0500 && descMap.attrInt == 0x0001 && device.getDataValue("model") == "RFMS-ZBMS") {
+				if (Integer.parseInt(descMap.value, 16) == 0x0015) {
+					//multi-sensor is in contact or tilt detector mode - both act as contact sensor so no action is necessary
+				} else if (Integer.parseInt(descMap.value, 16) == 0x002A) {
+					//multi-sensor is in water detector mode, DTH should be changed to water sensor DTH
+					log.debug "SmartSense Moisture Sensor"
+					setDeviceType("SmartSense Moisture Sensor")
```

Comment:
My understanding is when the device type is changed only `updated` will be called and not `configure`. That looks like it should be OK because `configure` does the same thing in both DTHs.

Reply:
@ALamchaS This explains why you have "Checking Status" when you switch the device type. The initial poll for device state is in `configure`. If you switch the device type then that won't happen. Maybe add an `updated` method to both DTHs that calls `refresh` and take the `refresh` call out of `configure`?

Reply:
This state persists even after I trigger wet/dry events, so it is not the reason for this problem.

I will try making water sensor the main DTH today. I will see if this situation gets inversed.
If it does, this is caused by dynamic DTH change,
if not - something may be wrong with this water sensor UI metadata.

Thanks for making me think of that.

Reply:
Ok. So after changing DTH situation is: water sensor works OK. contact sensor does not display correctly in OneApp.